dataland-csvconverter/src/main/kotlin/org/dataland/csvconverter/CsvToJsonConverter.kt

```diff
@@ -0,0 +1,222 @@
+import org.dataland.datalandbackend.model.CompanyAssociatedData
+import org.dataland.datalandbackend.model.CompanyIdentifier
+import org.dataland.datalandbackend.model.CompanyInformation
+import org.dataland.datalandbackend.model.EuTaxonomyData
+import org.dataland.datalandbackend.model.EuTaxonomyDetailsPerCashFlowType
+import org.dataland.datalandbackend.model.enums.AttestationOptions
+import org.dataland.datalandbackend.model.enums.IdentifierType
+import org.dataland.datalandbackend.model.enums.StockIndex
+import org.dataland.datalandbackend.model.enums.YesNo
+import java.io.File
+import java.io.FileReader
+import java.math.BigDecimal
+import java.nio.charset.StandardCharsets
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+/**
+ * Class to transform company information and EU Taxonomy data delivered by csv into json format
+ * @param filePath location of the csv file to be transformed
+ */
+class CsvToJsonConverter(private val filePath: String) {
+
+    private val objectMapper = ObjectMapper().registerModule(JavaTimeModule())
+        .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+    private val notAvailableString = "n/a"
+    private var euroUnitConverter = "1000000"
+    private val rawCsvData: List<Map<String, String>> = readCsvFile(filePath)
+
+    private val columnMapping = mapOf(
+        "companyName" to "Company name",
+        "headquarters" to "Headquarter",
+        "sector" to "Sektor",
+        "marketCap" to "Market Capitalization (EURmm)",
+        "reportingDateOfMarketCap" to "Market Capitalization Date",
+        "totalRevenue" to "Total Revenue in EURmio",
+        "totalCapex" to "Total CapEx EURmio",
+        "totalOpex" to "Total OpEx EURmio",
+        "eligibleRevenue" to "Eligible Revenue",
+        "eligibleCapex" to "Eligible CapEx",
+        "eligibleOpex" to "Eligible OpEx",
+        "alignedRevenue" to "Aligned Revenue",
+        "alignedCapex" to "Aligned CapEx",
+        "alignedOpex" to "Aligned OpEx",
+        "companyType" to "IS/FS",
+        "reportObligation" to "NFRD Pflicht",
+        "attestation" to "Assurance"
+    )
+
+    private val stockIndexMapping = mapOf(
+        StockIndex.PrimeStandard to "Prime Standard",
+        StockIndex.GeneralStandard to "General Standard",
+        StockIndex.Hdax to "HDAX",
+        StockIndex.Scale to "Scale",
+        StockIndex.Cdax to "CDAX",
+        StockIndex.Gex to "GEX",
+        StockIndex.Dax to "DAX",
+        StockIndex.Mdax to "MDAX",
+        StockIndex.Sdax to "SDAX",
+        StockIndex.TecDax to "TecDAX",
+        StockIndex.Dax50Esg to "DAX 50 ESG"
+    )
+
+    private val identifierMapping = mapOf(
+        IdentifierType.Isin to "ISIN",
+        IdentifierType.Lei to "LEI",
+        IdentifierType.PermId to "PermID"
+    )
+
+    private inline fun <reified T> readCsvFile(fileName: String): List<T> {
+        FileReader(fileName, StandardCharsets.UTF_8).use { reader ->
```

Comment:
sollen wir hier it statt reader benutzen? Finde ich ausreichend klar - und ist kürzer.

Reply:
ist geändert

Reply:
ok